### PR TITLE
Some quality of life improvements

### PR DESCRIPTION
**Pull request by poros**
I started to write documentation for the project and I decided to add some quality of life improvements for developers. I hope you don't mind, @flavray.

If you don't like the output of `cargo fmt`, I can add a `rustfmt.toml` with the configurations that you want to change.

### CODE REVIEW

**Comment by poros** [comment]
I added the configurations for rustfmt you were so kind to point out. I had to rebase since rustfmt is not  a bijective function, so the PR might look a bit weird. Have another look

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,46 @@
+VENV := precommit_venv
+HOOKS := .git/hooks/pre-commit
+
+# PRE-COMMIT HOOKS
+
+$(VENV): .requirements-precommit.txt
+	rm -rf $(VENV)
+	virtualenv -p python3.6 $(VENV)
+	$(VENV)/bin/pip install -i https://pypi.yelpcorp.com/simple/ -r .requirements-precommit.txt
```
Comment (flavray) [inline Makefile]: s/yelpcorp.com/python.org :)

File: src/codec.rs [hunk 2]
```diff
@@ -45,13 +50,14 @@ impl Codec {
...
                 let mut encoder = Encoder::new(Vec::new());
-                encoder.write(stream)?;
+                encoder.write_all(stream)?;
                 *stream = encoder.finish().into_result()?;
             },
-            #[cfg(feature = "snappy")] &Codec::Snappy => {
+            #[cfg(feature = "snappy")]
+            &Codec::Snappy => {
                 let mut writer = Writer::new(Vec::new());
                 writer.write(stream)?;
```
Comment (flavray) [inline src/codec.rs]: Given the change clippy made L53 of the same file, shouldn't this be `writer.write_all`?

Snap's `Writer` [implements the `io::Write`trait](https://docs.rs/snap/0.2.4/snap/struct.Writer.html#impl-Write), so I suppose that applies here as well?

Maybe clippy does not go inside feature code by default?

Comment (poros) [reply src/codec.rs]: Nice finding! You are correct, I have to pass to clippy the features the code has to be compiled with. I updated the Makefile and fixed the above issue for snappy as well.

File: .requirements-precommit.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+pre-commit==1.6.0
```
[the remark this answers is not in the document]
Comment (poros) [reply .requirements-precommit.txt]: I'm so used to pre-commit hooks that is just natural for me to have them in any project, but I understand your point. Up to you, anyway, I can remove it if you prefer so.

**Pull request merged**